src/compas_viewer/components/renderer/renderer.py

```diff
@@ -518,13 +519,21 @@ def sort_objects_from_category(
         tag_objs = []
         vector_objs = []
         mesh_objs = []
-        for obj in objs:
+
+        def sort(obj):
             if isinstance(obj, TagObject):
                 tag_objs.append(obj)
             elif isinstance(obj, VectorObject):
                 vector_objs.append(obj)
             else:
                 mesh_objs.append(obj)
+
+        for obj in objs:
+            if isinstance(obj, CollectionObject):
+                [sort(item) for item in obj.items]
```

Comment:
If you do this, doesn't it mean the objects of the collection are then drawn separately instead of the combined one, hence not benefit from the performance boost of using single combined buffer?

Reply:
We need this sorting function to match the category of the object and the corresponding shader. If the `CollectionObject` contains different objects, i.e. a mesh with a tag, the code will fail if we don't sort them.

Yet, this function is cached by `lru_cache`, meaning it will not be called every frame if the inputs are the same.

Reply:
I don't think you get my point yet. In the past in `view2`, the Collection object is not just a group. It was created more for performance reasons. When you have many separated objects, because they all have independent buffers, they are rendered sequentially rather than in parallel in GPU. The reason to have `CollectionObject` is to join seperate object buffers (points, lines, faces) into a one big buffer. That's what `_read_xxx_data` in `CollectionObject` is doing, so they can be feed into GPU in one go. We were also able to combine different object types because almost all objects inherented from `BufferObject` except `TextObject`. The way you are using it now is taking the sub-objects out and skipping the buffer combination.